addons/addon-base-raas/packages/base-raas-services/lib/data-egress/data-egress-service.js

```diff
@@ -537,6 +546,118 @@ class DataEgressService extends Service {
     return memberAccount.accountId;
   }
 
+  getMainAccountEgressStoreRole(egressStoreId) {
+    return `study-${egressStoreId}`;
+  }
+
+  getMainAccountEgressStoreRolePolicyName(egressStoreId) {
+    return `study-${egressStoreId}`;
+  }
+
+  /**
+   * Create the main account IAM Role  and Policy to allow a workspace on the member account to access
+   * the egress store S3 bucket
+   * @param requestContext
+   * @param egressStoreId
+   * @returns role Arn
+   */
+  async createMainAccountEgressStoreRole(requestContext, egressStoreId) {
+    const egressStoreBucketName = this.settings.get('egressStoreBucketName');
+    const kmsArn = await this.getKmsKeyIdArn();
+    const memberAccountId = await this.getMemberAccountId(requestContext, egressStoreId);
+    const mainAccountRoleName = this.getMainAccountEgressStoreRole(egressStoreId);
+    const permissionPolicy = {
```

Comment:
Can you look into [study-policy.js](https://github.com/awslabs/service-workbench-on-aws/blob/feat-secure-workspace-egress/addons/addon-base-raas/packages/base-raas-services/lib/helpers/iam/study-policy.js) and see if that works for you? Most likely it should, but in case it doesn't we can extend it if we need to rather than hardcoding the policy here.

Reply:
Updated to use study-policy